### PR TITLE
METRON-435 Create Stellar REPL

### PR DESCRIPTION
**Pull request by nickwallen**
Created this for debugging some issues I was having in running Stellar functions are on a live cluster.  Thought this would be generally useful on its own.

METRON-435

Create a REPL (Read Eval Print Loop) for the Stellar language that helps in debugging, troubleshooting and learning Stellar. The Stellar DSL (domain specific language) is used to act upon streaming data within Apache Storm. It is difficult to troubleshoot Stellar when it can only be executed within a Storm topology. This REPL is intended to help mitigate that problem by allowing a user to replicate data encountered in production, isolate initialization errors, or understand function resolution problems.

### CODE REVIEW

**Comment by cestella** [comment]
did you mean to title this METRON-435?

File: metron-platform/metron-common/src/main/java/org/apache/metron/common/stellar/shell/StellarExecutor.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+      try (CuratorFramework client = ConfigurationsUtils.getClient(zookeeperUrl)) {
+        client.start();
+
+        Map<String, Object> global = JSONUtils.INSTANCE.load(
+                new ByteArrayInputStream(readGlobalConfigBytesFromZookeeper(client)),
+                new TypeReference<Map<String, Object>>() {
+                });
+
+        context = new Context.Builder()
+                .with(Context.Capabilities.GLOBAL_CONFIG, () -> global)
```
Comment (cestella) [inline metron-platform/metron-common/src/main/java/org/apache/metron/common/stellar/shell/StellarExecutor.java]: Can we add the `Context.Capabilities.ZOOKEEPER_CLIENT` as well?  Looks like we have it right there on line 81.

File: metron-platform/metron-common/src/main/java/org/apache/metron/common/stellar/shell/StellarShell.java [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+import static java.lang.String.format;
+
+/**
+ * A REPL environment for Stellar.
+ *
+ * Useful for debugging Stellar expressions.
+ */
+public class StellarShell {
+
+  private static final String WELCOME = "Stellar, Go!";
```
Comment (cestella) [inline metron-platform/metron-common/src/main/java/org/apache/metron/common/stellar/shell/StellarShell.java]: I LOL'd

**Comment by nickwallen** [comment]
Oops, yes.  Let me fix.

File: metron-platform/metron-common/src/main/java/org/apache/metron/common/stellar/shell/StellarExecutor.java (hunk 1 above)
Comment (nickwallen) [inline metron-platform/metron-common/src/main/java/org/apache/metron/common/stellar/shell/StellarExecutor.java]: Added zookeeper client to the Context.

**Comment by nickwallen** [comment]
Also, added a doc string type functionality.  I couldn't help myself.

```
>>> ?PROTOCOL_TO_NAME
PROTOCOL_TO_NAME
 desc: Convert the IANA protocol number to the protocol name       
 args: IANA Number                                                 
  ret: The protocol name associated with the IANA number.  
```

**Comment by cestella** [comment]
What, no autocomplete? ;)

**Comment by cestella** [comment]
+1, we can iterate from here.  I think this is going to be super useful.

**Comment by nickwallen** [comment]
Ha, trust me, I wanted to do it.

**Comment by cestella** [comment]
Bonus points if you had used https://github.com/qmx/jreadline and given us emacs and vi mode as well as a hook for autocomplete :)

**Pull request closed without merging**